### PR TITLE
Add new tokens

### PR DESCRIPTION
**Pull request by marvin-bitterlich**
Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 25 issues
* ![MINOR][MINOR] 25 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [basic](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/entity/ai/basic): Add a 'package-info.java' file to document the 'basic' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [farmer](https://github.com/Minecolonies/minecolonies/blob/c2dabb90380c3678c82f83b94d5b29bf665867fb/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer): Add a 'package-info.java' file to document the 'farmer' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

File: src/api/java/com/minecolonies/api/colony/requestsystem/location/ILocationFactory.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Method to get a new instance of a location given the input.
+     *
+     * @param input The input to build a new location for.
+     * @return The new output instance for a given input.
+     */
+    @NotNull
+    L getNewInstance(@NotNull T input);
+}
```
Comment (Raycoms) [inline src/api/java/com/minecolonies/api/colony/requestsystem/location/ILocationFactory.java:45]: New line

**Comment by Raycoms** [comment]
:shipit: I reviewed all of those already a while ago.

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1081/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 25 issues
* ![MINOR][MINOR] 25 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [basic](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/entity/ai/basic): Add a 'package-info.java' file to document the 'basic' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [farmer](https://github.com/Minecolonies/minecolonies/blob/ccc5ebdced013ee32eca33aef11c4b8bac7a131d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer): Add a 'package-info.java' file to document the 'farmer' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**